diff --git a/build/config/ios/BUILD.gn b/build/config/ios/BUILD.gn
index 6a4195b1..6a98460 100644
--- a/build/config/ios/BUILD.gn
+++ b/build/config/ios/BUILD.gn
@@ -157,6 +157,7 @@
 
   ldflags += [ "-Wl,-rpath,@executable_path/Frameworks" ]
 
+  # TODO(crbug.com/424634516): Remove once Apple fixes the issue.
   if (xcode_version_int >= 2600) {
     framework_dirs = [ "$ios_sdk_path/System/Library/SubFrameworks" ]
   }
@@ -177,6 +178,11 @@
   }
 
   ldflags += [ "-Wl,-rpath,@executable_path/../../Frameworks" ]
+
+  # TODO(crbug.com/424634516): Remove once Apple fixes the issue.
+  if (xcode_version_int >= 2600) {
+    framework_dirs = [ "$ios_sdk_path/System/Library/SubFrameworks" ]
+  }
 }
 
 config("ios_dynamic_flags") {
@@ -229,6 +235,11 @@
     "-Wl,-rpath,@executable_path/Frameworks",
     "-Wl,-rpath,@loader_path/Frameworks",
   ]
+
+  # TODO(crbug.com/424634516): Remove once Apple fixes the issue.
+  if (xcode_version_int >= 2600) {
+    framework_dirs = [ "$ios_sdk_path/System/Library/SubFrameworks" ]
+  }
 }
 
 config("xctest_config") {
@@ -253,6 +264,7 @@
     "XCTest.framework",
   ]
 
+  # TODO(crbug.com/424634516): Remove once Apple fixes the issue.
   if (xcode_version_int >= 2600) {
     framework_dirs = [ "$ios_sdk_path/System/Library/SubFrameworks" ]
   }
