diff --git a/webrtc/media/webrtc/webrtcvideocapturer.cc b/webrtc/media/webrtc/webrtcvideocapturer.cc
index d152124..332fcb2 100644
--- a/webrtc/media/engine/webrtcvideocapturer.cc
+++ b/webrtc/media/engine/webrtcvideocapturer.cc
@@ -206,7 +206,7 @@ bool WebRtcVideoCapturer::Init(const Device& device) {
 
   if (supported.empty()) {
     LOG(LS_ERROR) << "Failed to find usable formats for id: " << device.id;
-    return false;
+//    return false;
   }
 
   module_ = factory_->Create(0, vcm_id);
diff --git a/webrtc/modules/video_capture/video_capture.gypi b/webrtc/modules/video_capture/video_capture.gypi
index c80f2bf..424b993 100644
--- a/webrtc/modules/video_capture/video_capture.gypi
+++ b/webrtc/modules/video_capture/video_capture.gypi
@@ -68,29 +68,6 @@
                 'linux/video_capture_linux.h',
               ],
             }],  # linux
-            ['OS=="mac"', {
-              'sources': [
-                'mac/qtkit/video_capture_qtkit.h',
-                'mac/qtkit/video_capture_qtkit.mm',
-                'mac/qtkit/video_capture_qtkit_info.h',
-                'mac/qtkit/video_capture_qtkit_info.mm',
-                'mac/qtkit/video_capture_qtkit_info_objc.h',
-                'mac/qtkit/video_capture_qtkit_info_objc.mm',
-                'mac/qtkit/video_capture_qtkit_objc.h',
-                'mac/qtkit/video_capture_qtkit_objc.mm',
-                'mac/qtkit/video_capture_qtkit_utility.h',
-                'mac/video_capture_mac.mm',
-              ],
-              'link_settings': {
-                'xcode_settings': {
-                  'OTHER_LDFLAGS': [
-                    '-framework Cocoa',
-                    '-framework CoreVideo',
-                    '-framework QTKit',
-                  ],
-                },
-              },
-            }],  # mac
             ['OS=="win"', {
               'dependencies': [
                 '<(DEPTH)/third_party/winsdk_samples/winsdk_samples.gyp:directshow_baseclasses',
@@ -133,31 +110,33 @@
                 },
               },
             }],
-            ['OS=="ios"', {
+            ['OS=="mac"', {
               'sources': [
-                'ios/device_info_ios.h',
-                'ios/device_info_ios.mm',
-                'ios/device_info_ios_objc.h',
-                'ios/device_info_ios_objc.mm',
-                'ios/rtc_video_capture_ios_objc.h',
-                'ios/rtc_video_capture_ios_objc.mm',
-                'ios/video_capture_ios.h',
-                'ios/video_capture_ios.mm',
+                'mac/device_info_mac.h',
+                'mac/device_info_mac.mm',
+                'mac/device_info_mac_objc.h',
+                'mac/device_info_mac_objc.mm',
+                'mac/rtc_video_capture_mac_objc.h',
+                'mac/rtc_video_capture_mac_objc.mm',
+                'mac/video_capture_mac.h',
+                'mac/video_capture_mac.mm',
               ],
               'xcode_settings': {
                 'CLANG_ENABLE_OBJC_ARC': 'YES',
               },
+              'cxxflags': [
+                '-Wno-partial-availability'
+              ],
               'all_dependent_settings': {
                 'xcode_settings': {
                   'OTHER_LDFLAGS': [
                     '-framework AVFoundation',
                     '-framework CoreMedia',
-                    '-framework CoreVideo',
-                    '-framework UIKit',
+                    '-framework CoreVideo'
                   ],
                 },
               },
-            }],  # ios
+            }],  # mac
           ], # conditions
         },
       ],
