Index: DEPS
===================================================================
--- DEPS	(revision 6937)
+++ DEPS	(working copy)
@@ -387,5 +387,9 @@
     "action": ["python", Var("root_dir") + "/webrtc/build/gyp_webrtc",
                Var("extra_gyp_flag")],
   },
+  {
+    "pattern": "tools/sanitizer_options/sanitizer_options.cc",
+    "action" : ["svn", "update", "-r", Var("chromium_revision"), Var("root_dir") + "/tools/sanitizer_options/sanitizer_options.cc"],
+  }
 ]
 
Index: webrtc/examples/android/media_demo/jni/jni_helpers.h
===================================================================
--- webrtc/examples/android/media_demo/jni/jni_helpers.h	(revision 6937)
+++ webrtc/examples/android/media_demo/jni/jni_helpers.h	(working copy)
@@ -20,6 +20,7 @@
 
 #include <map>
 #include <string>
+#include <stdlib.h>
 
 #define TAG "WEBRTC-NATIVE"
 
Index: webrtc/system_wrappers/interface/timestamp_extrapolator.h
===================================================================
--- webrtc/system_wrappers/interface/timestamp_extrapolator.h	(revision 6937)
+++ webrtc/system_wrappers/interface/timestamp_extrapolator.h	(working copy)
@@ -31,7 +31,7 @@
     bool DelayChangeDetection(double error);
     RWLockWrapper*        _rwLock;
     double                _w[2];
-    double                _P[2][2];
+    double                mP[2][2];
     int64_t         _startMs;
     int64_t         _prevMs;
     uint32_t        _firstTimestamp;
Index: webrtc/system_wrappers/source/timestamp_extrapolator.cc
===================================================================
--- webrtc/system_wrappers/source/timestamp_extrapolator.cc	(revision 6937)
+++ webrtc/system_wrappers/source/timestamp_extrapolator.cc	(working copy)
@@ -47,9 +47,9 @@
     _firstTimestamp = 0;
     _w[0] = 90.0;
     _w[1] = 0;
-    _P[0][0] = 1;
-    _P[1][1] = _P11;
-    _P[0][1] = _P[1][0] = 0;
+    mP[0][0] = 1;
+    mP[1][1] = _P11;
+    mP[0][1] = mP[1][0] = 0;
     _firstAfterReset = true;
     _prevUnwrappedTimestamp = -1;
     _prevWrapTimestamp = -1;
@@ -112,14 +112,14 @@
         // A sudden change of average network delay has been detected.
         // Force the filter to adjust its offset parameter by changing
         // the offset uncertainty. Don't do this during startup.
-        _P[1][1] = _P11;
+        mP[1][1] = _P11;
     }
     //T = [t(k) 1]';
     //that = T'*w;
     //K = P*T/(lambda + T'*P*T);
     double K[2];
-    K[0] = _P[0][0] * tMs + _P[0][1];
-    K[1] = _P[1][0] * tMs + _P[1][1];
+    K[0] = mP[0][0] * tMs + mP[0][1];
+    K[1] = mP[1][0] * tMs + mP[1][1];
     double TPT = _lambda + tMs * K[0] + K[1];
     K[0] /= TPT;
     K[1] /= TPT;
@@ -127,12 +127,12 @@
     _w[0] = _w[0] + K[0] * residual;
     _w[1] = _w[1] + K[1] * residual;
     //P = 1/lambda*(P - K*T'*P);
-    double p00 = 1 / _lambda * (_P[0][0] - (K[0] * tMs * _P[0][0] + K[0] * _P[1][0]));
-    double p01 = 1 / _lambda * (_P[0][1] - (K[0] * tMs * _P[0][1] + K[0] * _P[1][1]));
-    _P[1][0] = 1 / _lambda * (_P[1][0] - (K[1] * tMs * _P[0][0] + K[1] * _P[1][0]));
-    _P[1][1] = 1 / _lambda * (_P[1][1] - (K[1] * tMs * _P[0][1] + K[1] * _P[1][1]));
-    _P[0][0] = p00;
-    _P[0][1] = p01;
+    double p00 = 1 / _lambda * (mP[0][0] - (K[0] * tMs * mP[0][0] + K[0] * mP[1][0]));
+    double p01 = 1 / _lambda * (mP[0][1] - (K[0] * tMs * mP[0][1] + K[0] * mP[1][1]));
+    mP[1][0] = 1 / _lambda * (mP[1][0] - (K[1] * tMs * mP[0][0] + K[1] * mP[1][0]));
+    mP[1][1] = 1 / _lambda * (mP[1][1] - (K[1] * tMs * mP[0][1] + K[1] * mP[1][1]));
+    mP[0][0] = p00;
+    mP[0][1] = p01;
     _prevUnwrappedTimestamp = unwrapped_ts90khz;
     if (_packetCount < _startUpFilterDelayInPackets)
     {
