diff --git a/lib/tls/openssl/openssl-session.c b/lib/tls/openssl/openssl-session.c
index e27a8f3..47edecc 100644
--- a/lib/tls/openssl/openssl-session.c
+++ b/lib/tls/openssl/openssl-session.c
@@ -235,7 +235,7 @@ lws_tls_session_new_cb(SSL *ssl, SSL_SESSION *sess)
 
 	/* api return is long, although we only support setting
 	 * default (300s) or max uint32_t */
-	ttl = SSL_SESSION_get_timeout(sess);
+	ttl = (long)SSL_SESSION_get_timeout(sess);
 
 	lws_context_lock(vh->context, __func__); /* -------------- cx { */
 	lws_vhost_lock(vh); /* -------------- vh { */
diff --git a/lib/tls/openssl/openssl-x509.c b/lib/tls/openssl/openssl-x509.c
index dac4aa3..a1b2ffd 100644
--- a/lib/tls/openssl/openssl-x509.c
+++ b/lib/tls/openssl/openssl-x509.c
@@ -248,7 +248,11 @@ lws_tls_openssl_cert_info(X509 *x509, enum lws_tls_cert_info type,
 		{
 			const X509V3_EXT_METHOD* method = X509V3_EXT_get(ext);
 			STACK_OF(CONF_VALUE) *cv;
+#if defined(LWS_WITH_BORINGSSL)
+			size_t j;
+#else
 			int j;
+#endif
 
 			cv = i2v_GENERAL_NAMES((X509V3_EXT_METHOD*)method, akid->issuer, NULL);
 			if (!cv)
